### ISSUE TITLE
Is it possible to use CarrierWave instead of ActiveStorage?

### ISSUE THREAD

**Issue by lukaVarga** [SpinaCMS/Spina#404]
Pretty self explanatory - is it possible to use CarrierWave (like in Spina 0.x.x) instead of ActiveStorage for image handling via Spina?

Issues with AS:
- every image request gets sent to the app
- app must find the url of each image (eg AWS's), triggering additional DB requests
- app must generate a "secure" url (considering blogs are usually public, this is completely pointless)
- one cannot use CDN in front of it
- the generated URL is not SEO friendly

So basically, we're bleeding performance, increasing load on our servers, generating unfriendly URLs for images and cannot use CDNs to improve performance. 

With CarrierWave, everything was working out of the box. At the moment, I don't really see the benefit with using AS for a public blog use-case.

So, is there anything we can do or are we stuck with AS and monkey-patching (both Spina and Rails) in case we wish to use Spina 1.x.x?

**Comment by Bramjetten** [comment]
I get where you're coming from. In an effort to minimize our dependencies, we – perhaps prematurely – switched to ActiveStorage.

This alternative RepresentationsController is a nice bandaid for now:

```ruby
# frozen_string_literal: true

module ActiveStorage
  class RepresentationsController < BaseController
    include ActiveStorage::SetBlob

    def show
      expires_in 1.year, public: true
      variant = @blob.representation(params[:variation_key]).processed
      send_data @blob.service.download(variant.key),
                type: @blob.content_type || DEFAULT_SEND_FILE_TYPE,
                disposition: 'inline'
    end
  end
end
```

**Comment by lukaVarga** [comment]
@Bramjetten thanks for the quick response! 

This will only make the URL public, right? It won't help with additional DB queries, non-SEO friendly urls and other issues?
I suppose I could monkey-patch it further and at least redirect it to use CDN, but that's still far from perfect.
At this point, I am actually considering manually uploading images and manually adding links to templates, so instead of using Spina::Image, I'd be using Spina::Text... but that's sooo 🤢

**Comment by Bramjetten** [comment]
With this controller you can safely use a CDN and still use all goodies ActiveStorage provides. The only thing this doesn't tackle is your SEO-friendly URLs, but that's mostly related to filenames imo.

**Comment by lukaVarga** [comment]
Thanks, I'll try it out :)
